src/main/java/com/google/sps/servlets/EditCardServlet.java

```diff
@@ -0,0 +1,118 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import com.google.gson.Gson;
+import com.google.sps.tool.ResponseSerializer;
+import java.util.Map;
+import java.util.List;
+
+@WebServlet("/editcard")
+public class EditCardServlet extends HttpServlet {
+
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    if (userService.isUserLoggedIn()) {
```

Comment:
Avoid indentation when possible. I would prefer to write this as:

```js
if (!userService.isUserLoggedIn()) {
  // send error ... ?
  return;
}

String cardKey = request.getParameter("cardKey");
...
```

Reply:
done